extensions/darwin/clock-nanosleep.c

```diff
@@ -0,0 +1,80 @@
+/* Copyright (C) 2020-2022 Aleix Conchillo Flaqué <aconchillo@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA
+ */
+
+#include "clock-nanosleep.h"
+
+#include <errno.h>
+#include <time.h>
+#include <sys/types.h>
+
+/**
+ * The code below has been extracted from the following notcurses files:
+ *
+ * https://github.com/dankamongmen/notcurses
+ *
+ *   src/compat/compat.c
+ *   src/compat/compat.h
+ */
+
+#define NANOSECS_IN_SEC 1000000000ul
+
+#ifdef __APPLE__
+#define TIMER_ABSTIME 1
+#endif
```

Comment:
I think this should be something like

```
#ifndef TIMER_ABSTIME
#ifdef __APPLE__
#define TIMER_ABSTIME 1
#else
#error TIMERS_ABSTIME undefined
#endif
```

instead. Otherwise, if Apple ever defines TIMER_ABSTIME properly, it would now be defined twice, which can lead to compiler warnings (or errors, depending on how strict it is), and bugs if the numeric value is changed.

Reply:
nice. done!